lib/api/controllers/authController.js

```diff
@@ -91,4 +109,37 @@ module.exports = function AuthController (kuzzle) {
         return q.reject(new ResponseObject(modifiedRequestObject || requestObject, invalid));
       });
   };
+
+  /**
+   * Updates the current user if it is not anonymous
+   *
+   * @param {RequestObject} requestObject
+   * @param context
+   * @returns {Promise}
+   */
+  this.selfUpdate = function (requestObject, context) {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('security:beforeSelfUpdate', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+        if (!context.token._id) {
+          return q.reject(new UnauthorizedError('User must be connected in order to call selfUpdate'));
+        }
+
+        if (modifiedRequestObject.data.body.profile) {
+          return q.reject(new BadRequestError('profile can not be provided in selfUpdate'));
+        }
+
+        if (modifiedRequestObject.data.body._id) {
+          return q.reject(new BadRequestError('_id can not be part of the body'));
+        }
+
+        return kuzzle.repositories.user.load(context.token.user._id);
+      })
+      .then(user => kuzzle.repositories.user.persist(_.extend(user, modifiedRequestObject.data.body), { database: { method: 'update' } }))
```

Comment:
the difference btw. _.extend and _.merge is only about child objects. _.extend will replace child objects, as _.merge will merge them.
see http://stackoverflow.com/questions/19965844/lodash-difference-between-extend-assign-and-merge.

example:
console.log(_.extend({a:{x: '1'}},{a:{y: '2'}}));
> { a: { y: '2' } }
console.log(_.merge({a:{x: '1'}},{a:{y: '2'}}));
> { a: { x: '1', y: '2' } }

In our case, I wonder if merge should be a better choice: if we have a user with nested attribute a.x = 1, and we want to update him with a new nested attribute a.y = 2, we still want to keep a.x, I guess...
(so it means that we should also fix the securityController as well, maybe...)